src/framework/components/render/component.js

```diff
@@ -0,0 +1,823 @@
+        }
+    },
+
+    /**
+     * @private
+     * @function
+     * @name pc.RenderComponent#show
+     * @description Enable rendering of the render {@link pc.MeshInstance}s if hidden using {@link pc.RenderComponent#hide}.
+     * This method sets all the {@link pc.MeshInstance#visible} property on all mesh instances to true.
+     */
+    show: function () {
+        if (this._meshInstances) {
+            for (var i = 0; i < this._meshInstances.length; i++) {
+                this._meshInstances[i].visible = true;
+            }
+        }
+    },
+
+    _onRenderAssetAdded: function () {
+        if (!this._assetReference.asset) return;
+
+        if (this._assetReference.asset.resource) {
+            this._onRenderAssetLoad();
+        } else if (this.enabled && this.entity.enabled) {
+            this.system.app.assets.load(this._assetReference.asset);
+        }
+    },
+
+    _onRenderAssetLoad: function () {
+
+        // remove existing instances
+        this.destroyMeshInstances();
+
+        var render = this._assetReference.asset.resource;
+        render.off('set:meshes', this._onSetMeshes, this);
+        render.on('set:meshes', this._onSetMeshes, this);
+        if (render.meshes) {
+            this._onSetMeshes(render.meshes);
+        }
+    },
+
+    _onSetMeshes: function (meshes) {
+        this._cloneMeshes(meshes);
+    },
+
+    _clearSkinInstances: function () {
+
+        for (var i = 0; i < this._meshInstances.length; i++) {
+            this._meshInstances[i].skinInstance = null;
+        }
+    },
+
+    _cloneSkinInstances: function () {
+
+        if (this._meshInstances.length && this._rootBone instanceof GraphNode) {
+
+            var j, skin, skinInst;
+
+            // maps each unique original skin to cloned skin instance
+            var skins = new Map();
+
+            for (var i = 0; i < this._meshInstances.length; i++) {
+                var meshInstance = this._meshInstances[i];
+                var mesh = meshInstance.mesh;
+
+                // if skinned but does not have instance created yet
+                if (mesh.skin && !mesh.skinInstance) {
+
+                    skin = mesh.skin;
+                    skinInst = skins.get(skin);
+
+                    // don't have skin instance for this skin
+                    if (!skinInst) {
+
+                        skinInst = new SkinInstance(skin);
+                        skins.set(skin, skinInst);
+
+                        // Resolve bone IDs to actual graph nodes
+                        var bones = [];
+                        for (j = 0; j < skin.boneNames.length; j++) {
+                            var boneName = skin.boneNames[j];
+                            var bone = this._rootBone.findByName(boneName);
+
+                            if (!bone) {
+                                console.error("Failed to find bone [", boneName, "] in the entity hierarchy, RenderComponent on ", this.entity.name, ", rootBone: ", this._rootBone.name);
```

Comment:
these should probably be concatenated with + instead of , ?

Reply:
done, thanks